### DISCUSSION TITLE
Fuzzy completion for `sf`!

### DISCUSSION THREAD

**Discussion by cristiand391** [forcedotcom/cli#2569] [Show and tell]
Hey folks 👋🏼 

Just wanted to show a plugin I wrote few weeks ago to add support for fuzzy completion of certain commands/flags via fzf's completion API.

repo: https://github.com/cristiand391/sf-plugin-fzf-cmp

Last year I wrote this small zsh script to improve command completion (most commands have >= 3 words, even with native shell completion you still have to type them in the right order): https://gist.github.com/cristiand391/7b3e2e51e3c427218d9dcf8ee0f7e828

[fuzzy-completion-demo.webm](https://github.com/cristiand391/sf-plugin-fzf-cmp/assets/6853656/a937d23b-4e63-41a0-8c2e-5676b8085b91)

fzf's completion only supports bash and zsh but this plugin doesn't work on bash yet.

After using it for 2 months the ones I use the most are [config vars](https://github.com/cristiand391/sf-plugin-fzf-cmp#configuration-variables), [commands](https://github.com/cristiand391/sf-plugin-fzf-cmp#commands) and [org flag completion](https://github.com/cristiand391/sf-plugin-fzf-cmp#org-usernames), but I'm happy to discuss adding support for new flag/commands :)

**Comment by cristiand391** [comment]
It's also worth learning fzf's search syntax:
https://github.com/junegunn/fzf#search-syntax

say you have a scratch org that you know its alias contains the word `bug`, you can just type `'bug` and fzf will only suggest the ones that contain that word (and if there's just one match it will insert it in the current line!): 

https://github.com/forcedotcom/cli/assets/6853656/6da501e4-5d3f-46b8-bb93-ba57d7a57a34

**Comment by AllanOricil** [comment]
Look at fig.io

**Reply by cristiand391** [reply]
Doesn't work on linux (and last I checked there was no huge updates after it was acquired by AWS).

**Reply by AllanOricil** [reply]
they rebranded to CodeWhisperer. Maybe it supports linux. I don't know. I'm still using the older figma because Im not interested in the code whisperer features

**Reply by AllanOricil** [reply]
It is not. just macos for now
<img width="723" alt="image" src="https://github.com/forcedotcom/cli/assets/55927613/b0927b13-99bc-412f-9c66-b98acfa7e6d1">

**Reply by AllanOricil** [reply]
Linux version is in close beta
https://fig.io/user-manual/linux
